### ISSUE TITLE
Enhancement: [no-unnecessary-type-assertion] Transitively check type assertion

### ISSUE THREAD

**Issue by SanderRonde** [typescript-eslint/typescript-eslint#9941]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-unnecessary-type-assertion

### Description

I propose that the `no-unnecessary-type-assertion` rule should also transitively check casts when multiple casts are chained. The pattern of chaining casts `(x as unknown as number)` is quite a common one for cases where the base type and the cast type don't overlap. Being able to spot these too would be an improvement imo.

### Fail

```TypeScript
const foo = 1;

function takesNumber(num: number) {
    // ...
}

takesNumber(foo as unknown as number);
```

### Pass

```TypeScript
const foo = 1;

function takesNumber(num: number) {
    // ...
}

takesNumber(foo);
```

### Additional Info

This is relatively easy to do, just go down the `.expression` property every time the expression itself is also a cast instead of only doing `.expression` once ([code](https://github.com/typescript-eslint/typescript-eslint/blob/1e5e9ea4cac25947c3a8748647a4fb4d329c4b25/packages/eslint-plugin/src/rules/no-unnecessary-type-assertion.ts#L258)). I've already implemented a version of this rule with this change in our own repository so I'd be open to submitting the PR myself too.

**Comment by bradzacher** [comment]
This looks like a duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/8107
